backend/plan/views.py

```diff
@@ -176,6 +183,106 @@ def recommend_courses_view(request):
         status=status.HTTP_200_OK,
     )
 
+class PrimaryScheduleViewSet(viewsets.ViewSet):
+    model = PrimarySchedule
+    queryset = PrimarySchedule.objects.none()
+    http_method_names = ["get", "post"]
+    permission_classes = [IsAuthenticated]
+    serializer_class = PrimaryScheduleSerializer
+
+    def get_queryset(self):
+        return PrimarySchedule.objects.all()
+
+    # TODO: generate proper PcxAutoschema for this viewset
+    schema = PcxAutoSchema(
+        response_codes={
+            reverse_func("primary-schedule", args=['friend_id', 'schedule_id']): {
+                "GET": {
+                    200: "Primary schedule (and friend's schedules) retrieved successfully.",
+                    400: "Invalid friend in request.",
+                    404: "Friend's schedule not found.",
+                },
+                "POST": {
+                    200: "Primary schedule updated successfully.",
+                    400 : "Invalid schedule in request.",
+                    403: "User does not have permission to use thiss schedule.",
+                }
+            },
+        },
+    )
+
+    def get(self, request):
+        res = {}
+        # return user primary schedule and all shared schedules with this user 
+        # if friend is passed in, then we return that specific friend's shared schedule
```

Comment:
Do we really need this functionality (filtering for a specific friend)? If not, we can just delete this whole method. Might be worth asking Andy/frontend. Less code is better

Reply:
I don't think so. If they need to filter by friend, they can just go through the primary schedules that they will already have loaded on the frontend and filter on "person" field there. Just going to delete the get method then.